From 6cb2f607e894ca6b8e41fe5bcef7a08157a8d8d0 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Wed, 27 Jul 2016 09:46:06 -0700
Subject: [PATCH 695/827] drm: xilinx: dp: Use signed type as the value can be negative

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Reported-by: Julia Lawall <julia.lawall@lip6.fr>
Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
(cherry picked from commit 668c9f98eec45affbb9c69cc50b17163a61fb9d4)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_dp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_dp.c b/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
index 1da915a..6c1c292 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
@@ -999,8 +999,8 @@ static int xilinx_drm_dp_mode_configure(struct xilinx_drm_dp *dp, int pclock,
 	u8 max_lanes = dp->link_config.max_lanes;
 	u8 max_link_rate_code = drm_dp_link_rate_to_bw_code(max_rate);
 	u8 bpp = dp->config.bpp;
-	u8 lane_cnt, i;
-	s8 clock;
+	u8 lane_cnt;
+	s8 clock, i;
 
 	for (i = ARRAY_SIZE(bws) - 1; i >= 0; i--) {
 		if (current_bw && bws[i] >= current_bw)
-- 
2.9.3

